core/src/main/scala/org/http4s/Query.scala

```diff
@@ -17,7 +17,7 @@ import scala.collection.mutable.ListBuffer
   * When rendered, the resulting `String` will have the pairs separated
   * by '&' while the key is separated from the value with '='
   */
-final class Query private (pairs: Vector[KeyValue])
+final class Query private (pairs: Vector[KeyValue], queryEncodeParams: Query.QueryEncodeParams = Query.QueryEncodeParams.Default)
```

Comment:
This could be kept bincompat with a secondary constructor, although it's private anyway.

Reply:
A ramification of this is configurability is that two queries that are equal can render to unequal values.